Civi/RemoteParticipant/RegistrationEventFactory.php

```diff
@@ -114,15 +114,30 @@ private function handleAdditionalParticipants(RegistrationEvent $registrationEve
         $additionalParticipantsProfile = \CRM_Remoteevent_RegistrationProfile::getRegistrationProfile(
           $event['event_remote_registration.remote_registration_additional_participants_profile']
         );
+        $currentAdditionalParticipantCount = 0;
         foreach ($additionalContactsData as $participantNo => &$contactData) {
+            $currentAdditionalParticipantCount++;
             $additionalParticipantsProfile->modifyContactData($contactData);
             $contactData['contact_type'] ??= 'Individual';
             $contactData['xcm_profile'] = $event['event_remote_registration.remote_registration_additional_participants_xcm_profile'];
             $additionalParticipantsData[$participantNo]['role_id'] ??= $this->getDefaultRoleId($event);
             $additionalParticipantsData[$participantNo]['event_id'] = $submissionData['event_id'];
 
+            // Check for waitlist.
+            if (
+                !empty($event['max_participants'])
+                && \CRM_Remoteevent_Registration::getRegistrationCount($event['id'])
```

Comment:
It probably makes sense to do this check at last because the other ones are just simple array lookups.